### PR TITLE
Add price currency to conferences

### PR DESCRIPTION
**Pull request by ed1001**
**Checklist**

- [x] I have read the [Contribution & Best practices Guide](https://github.com/openSUSE/osem/blob/master/CONTRIBUTING.md).
- [x] My branch is up-to-date with the upstream `master` branch.
- [ ] The tests pass locally with my changes.
- [x] I have added necessary documentation (if appropriate).

N.B. As I mentioned last pull request, I'm unclear how the test can be passed. The untouched master branch fails the test also so not sure what's wrong there. 

Ticket currencies - #2086

Declare the price currency of ticket when creating a conference

Add price currency to conferences and make amendments to pertinent functionality and views

- Added migration to add price_currency to conferences table
- Added select currency in form for new conference under basic info
- Changed with_model_currency: to with_currency: with a lambda that returns parent conference's currency
- Added price_currency to strong params of conference
- Created data rake task for setting existing conferences' price currency

Pull request one of two - as suggested I will hold off the removal of the price_currency from the tickets table until this is approved. Also will amend all price currency references in the various methods it's used in.

### CODE REVIEW

**Comment by hennevogel** [comment]
So this makes #2547 obsolete? If yes please close it.

**Comment by hennevogel** [comment]
> The untouched master branch fails the test also so not sure what's wrong there.

You should rebase, this has been fixed.

**Comment by ed1001** [comment]
Ok, closed #2547.

Will pull again and retest.

**Comment by ed1001** [comment]
I removed the price_currency from tickets and modified all refs to go through conference. Now I have a problem when running tests, in that I get a invalid factory error, but I'm not sure of the origin:

physical_ticket - undefined method `price_currency' for nil:NilClass (NoMethodError)

I'm having trouble finding what should be tweaked here, shall I commit another pull request so it can be reviewed?

**Comment by differentreality** [comment]
@ed1001 The factory for the ticket is here https://github.com/openSUSE/osem/blob/master/spec/factories/tickets.rb
But your error also mentions something about a nil class

Please rebase to
- get the fix for the features tests
- get the global changes for schema.rb so that your diff can merely include the meaningful changes you are actually introducing

**Comment by ed1001** [comment]
I did rebase already 4 days ago to get the test fix.

I already have been looking at the factories and didn't know why they were failing, this is my first experience with factories so forgive me for not catching the problem right away. I will just keep digging to find the issue.

**Comment by hennevogel** [comment]
@differentreality is this for GSoC? Shouldn't this go to the branch then?

**Comment by differentreality** [comment]
> @differentreality is this for GSoC? Shouldn't this go to the branch then?

Nop, it is not for GSoC

It addresses 
https://github.com/openSUSE/osem/issues/2086 as per Ana's comments

File: app/views/admin/conferences/new.html.haml [hunk 1]
```diff
@@ -7,6 +7,7 @@
                           input_html: { required: 'required' }
         = f.input :short_title, hint: "A short and unique handle for your conference, using only letters, numbers, underscores, and dashes. This will be used to identify your conference in URLs etc. Example: 'froscon2011'",
                                 input_html: { required: 'required', pattern: '[a-zA-Z0-9_-]+', title: 'Only letters, numbers, underscores, and dashes.' }, prepend: conferences_url + '/'
+        = f.input :price_currency, as: :select, class: 'form-control', collection: ['USD', 'EUR', 'GBP', 'INR', 'CNY', 'CHF'], include_blank: false, hint: "Please select the currency in which the tickets for this conference will be sold"
```
Comment (differentreality) [inline app/views/admin/conferences/new.html.haml]: The currency collection used to be in the dotenv file, wondering what happened to it!

**Comment by hennevogel** [comment]
Closing as stale. Ping us if you want to still do this. Sorry for the long wait...

**Pull request closed without merging**